examples/docs_beta_snippets/docs_beta_snippets/guides/components/shell-script-component/with-build-defs.py

```diff
@@ -21,18 +22,17 @@ class ShellScriptSchema(ResolvableSchema):
     asset_specs: Sequence[AssetSpecSchema]
 
 
+def resolve_asset_specs(
+    context: ResolutionContext, schema: ShellScriptSchema
+) -> Sequence[dg.AssetSpec]:
+    return context.resolve_value(schema.asset_specs)
+
+
 @registered_component_type(name="shell_command")
 @dataclass
 class ShellCommand(Component):
     script_path: str
-    asset_specs: Sequence[dg.AssetSpec]
-
-    @field_resolver("asset_specs")
-    @staticmethod
-    def resolve_asset_specs(
-        schema: ShellScriptSchema, context: ResolutionContext
-    ) -> Sequence[dg.AssetSpec]:
-        return context.resolve_value(schema.asset_specs)
+    asset_specs: Annotated[Sequence[dg.AssetSpec], FieldResolver(resolve_asset_specs)]
```

Comment:
I think we shoudl have a specialized name for this and bikeshed it but we can do that later

Reply:
yeah I think this ties in with your other comment -- basically how much automatic type conversion are we comfortable with. i opted to write the doc in such a way that they make any conversion manually (even though the framework doesn't require that at the moment) just to make everything very explicit. but we will likely walk that back or provide simpler syntax for these cases in the near future